api/v1alpha1/podhttpchaos_types.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2020 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// PodHttpChaosSpec defines the desired state of IoChaos
+type PodHttpChaosSpec struct {
+	// ProxyPorts represents the target ports to be proxy of.
+	ProxyPorts []int32 `json:"proxy_ports"`
+
+	// Pid represents a running tproxy process id
+	// +optional
+	Pid int64 `json:"pid,omitempty"`
+
+	// StartTime represents the start time of a tproxy process
+	// +optional
+	StartTime int64 `json:"start_time,omitempty"`
+
+	// Rules are a list of injection rule for http request
+	// +optional
+	Rules []*PodHttpChaosRule `json:"rules,omitempty"`
+}
+
+// PodHttpChaosRule defines the injection rule for http request
+type PodHttpChaosRule struct {
+	Target   PodHttpChaosTarget   `json:"target"`
+	Selector PodHttpChaosSelector `json:"selector"`
+	Actions  PodHttpChaosActions  `json:"actions"`
+}
+
+type PodHttpChaosSelector struct {
+	// +optional
+	Port *int32 `json:"port,omitempty"`
+
+	// +optional
+	Path *string `json:"path,omitempty"`
+
+	// +optional
+	Method *string `json:"method,omitempty"`
+
+	// +optional
+	Code *int32 `json:"code,omitempty"`
+
+	// +optional
+	RequestHeaders map[string]string `json:"request_headers,omitempty"`
```

Comment:
how about adding a comment, what is the key and value?

Reply:
Yes, it needs some comments.

Reply:
Comments added.